### PR TITLE
Update installation page for v1.0

### LINKED ISSUE #1393: Update installation docs
The [installation page](https://tskit.dev/msprime/docs/latest/installation.html) needs a refresh for 1.0

- [ ] Change references to Python 3.6 to 3.7
- [ ] Update pip installation instructions to remove requirement on GSL
- [ ] Add a subsection to the pip section "Build from source". Move the information about GSL into this section. Give it a label so that we can refer to this information about GSL from the development page, also.
- [ ] Refresh the conda section with output from a recent conda transation, and update the example to use more modern APIs (i.e., use sim_ancestry and ``ts.draw_text()``)

I'm not totally sold on the "Via" prefix to the sections here, maybe we could rename these to "Conda", "Pip" and "Containers"?

### PR DESCRIPTION
**Pull request by castedo**
Fixes #1393

### CODE REVIEW

File: docs/installation.md [hunk 1]
```diff
@@ -15,7 +15,7 @@ There are three options for installing `msprime`:
 
 Pre-built binary packages for `msprime` are available through
 [conda](<https://conda.io/docs/>), and built using [conda-forge](<https://conda-forge.org/>).
-Packages for Python 3.6, 3.7 and 3.8 are available for Linux, OSX and Windows.
+Packages for Python 3.7+ are available for Linux, OSX and Windows.
```
Comment (benjeffery) [inline docs/installation.md:18]: This section implies that conda is the only place to get the pre-built binaries, I know it is covered later on, but good to mention here.

**Review by benjeffery — APPROVED** [review]
Thanks @castedo! Just one comment.

File: docs/installation.md (hunk 1 above)
Comment (castedo) [reply docs/installation.md]: Good point. I've now improved the wording.

**Review by jeromekelleher — APPROVED** [review]
LGTM, thanks @castedo

**Pull request merged**